### DISCUSSION TITLE
Is WiFi.setHostname() implemented?

### DISCUSSION THREAD

**Discussion by fapplin** [earlephilhower/arduino-pico#967] [Q&A]
Hi. I saw some discussion on the MicroPython board about them fixing the implementation of the setHostName() method. It doesn't seem to work with this core. Is there any information about this?

**Comment by fapplin** [comment]
Was this questioned ever answered about setting the host name? What I'm after is to be able to change how the Pico shows up in the WIFI scan. It shows up as PICOW and I'd like to change that to something else.

**Reply by earlephilhower** [reply]
`setHostname` works for me.  Make sure to call it before ``WiFi.begin()` and be aware many routers will cache the DCHP lease/name so if you change the name you might need to have the router drop its cache.  For example, I just ran the following:

````
#include <WiFi.h>
void setup() {
  WiFi.setHostname("lopsided");
  WiFi.begin(STASSID, STAPSK);
}

void loop() {
}
````

![image](https://user-images.githubusercontent.com/11875/213000893-e8ad4f82-71c0-4aa2-a68c-7709676ea44f.png)

**Reply by fapplin** [reply]
I'll give it a try. Thanks.

**Reply by fapplin** [reply]
So, I'm doing:

```
WiFi.mode(WIFI_AP);
WiFi.setHostname("XXX");
WiFi.begin(p_ssid, p_password);
 ```
 Does doing Access Point make a difference?

**Reply by earlephilhower** [reply]
Yup, completely different code path.  STA == send a DHCP request with my name.  AP = I *am* the DHCP server and DNS server.

How exactly are you finding the name of the AP itself?  It's not going to register with anyone since it's lord of the network...

**Reply by fapplin** [reply]
So, I'm using my Android phone or tablet to connect with and it always shows up as PICOW no matter what I put in the code.

**Reply by fapplin** [reply]
Is this how I'd have to accomplish it?

https://github.com/esp8266/Arduino/blob/master/libraries/DNSServer/examples/DNSServer/DNSServer.ino

**Reply by earlephilhower** [reply]
Sorry, I don't quite grok.  What exactly is it showing as PICOW?  The SSID you're connecting to?  You need to set p_ssid to the SSID you want to broadcast, not the hostname.  I'm not sure how the phone would even be able to so a reverse DNS to find who 192.168.0.1 (i.e. the Pico's IP) is...

**Reply by earlephilhower** [reply]
The example is if you want name lookup, yes.  Normally you don't care so much what the name of your router (in this case, the Pico) is...

**Reply by fapplin** [reply]
OH!! Duh! Thanks you again!
